cores/esp32/esp32-hal-misc.c

```diff
@@ -42,31 +44,47 @@ void yield()
     vPortYield();
 }
 
-static uint32_t _cpu_freq_mhz = 240;
+static uint32_t _cpu_freq_mhz = CONFIG_ESP32_DEFAULT_CPU_FREQ_MHZ;
+static uint32_t _sys_time_multiplier = 1;
 
-bool cpuFrequencySet(uint32_t cpu_freq_mhz){
-    if(_cpu_freq_mhz == cpu_freq_mhz){
+bool setCpuFrequency(uint32_t cpu_freq_mhz){
+    rtc_cpu_freq_config_t conf, cconf;
+    rtc_clk_cpu_freq_get_config(&cconf);
+    if(cconf.freq_mhz == cpu_freq_mhz && _cpu_freq_mhz == cpu_freq_mhz){
         return true;
     }
-    rtc_cpu_freq_config_t conf;
     if(!rtc_clk_cpu_freq_mhz_to_config(cpu_freq_mhz, &conf)){
         log_e("CPU clock could not be set to %u MHz", cpu_freq_mhz);
         return false;
     }
-    rtc_clk_cpu_freq_set_config(&conf);
+#if ARDUHAL_LOG_LEVEL >= ARDUHAL_LOG_LEVEL_INFO
+    log_i("%s: %u / %u = %u Mhz", (conf.source == RTC_CPU_FREQ_SRC_PLL)?"PLL":((conf.source == RTC_CPU_FREQ_SRC_APLL)?"APLL":((conf.source == RTC_CPU_FREQ_SRC_XTAL)?"XTAL":"8M")), conf.source_freq_mhz, conf.div, conf.freq_mhz);
+    delay(10);
+#endif
+    rtc_clk_cpu_freq_set_config_fast(&conf);
     _cpu_freq_mhz = conf.freq_mhz;
+    _sys_time_multiplier = 80 / getApbFrequency();
```

Comment:
This will always set `_sys_time_multiplier` to 0 all the time?

Reply:
how did I miss this :D